tokio/src/runtime/tests/loom_basic_scheduler.rs

```diff
@@ -0,0 +1,55 @@
+use crate::loom::sync::atomic::AtomicUsize;
+use crate::loom::sync::Arc;
+use crate::runtime;
+use crate::sync::oneshot::{self, Receiver};
+use crate::task;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::atomic::Ordering::{Acquire, Release};
+use std::task::{Context, Poll};
+
+#[test]
+fn block_on_num_polls() {
+    loom::model(|| {
+        let rt = runtime::Builder::new_current_thread().build().unwrap();
+
+        let (tx, rx) = oneshot::channel();
+        let num_polls = Arc::new(AtomicUsize::new(0));
```

Comment:
Loom tests are not really interesting when there's only one thread, however I'd like to see a test that spawns a few threads, doing the `spawn`/`block_on` from below in each thread, but each instance of the test performed on the same runtime shared with `Arc<Runtime>`.

Reply:
It would also make sense to have just the above test without threading, but it should be a non-loom test.

Reply:
Good suggestion, thanks a lot.